### PR TITLE
280 sim contractrunner from address should be accessed with dot notation

### PR DESCRIPTION
**Pull request by cristiam86**
# What
The contract_runner helper has been refactored to a class so developers can access its attributes such as `from_address` with dot notation instead of dict notation (square brackets and double quotes)

### CODE REVIEW

File: backend/node/genvm/base.py [hunk 1]
```diff
@@ -13,6 +13,16 @@
 from backend.node.genvm.code_enforcement import code_enforcement_check
 
 
+class ContractRunner:
+    def __init__(self, mode: str, node_config: dict):
+        self.mode = mode
```
Comment (AgustinRamiroDiaz) [inline backend/node/genvm/base.py:18]: I'm still not familiar entirely with the codebase, so just to learn: Would it be better to have 2 types of `ContractRunners`, one for validators and another one for leaders?

File: backend/node/genvm/base.py [hunk 2]
```diff
@@ -13,6 +13,16 @@
 from backend.node.genvm.code_enforcement import code_enforcement_check
 
 
+class ContractRunner:
```
Comment (AgustinRamiroDiaz) [inline backend/node/genvm/base.py:16]: Could you add comments explaining what each field represents please?

File: backend/node/genvm/base.py [hunk 3]
```diff
@@ -13,6 +13,16 @@
...
+    def __init__(self, mode: str, node_config: dict):
+        self.mode = mode
+        self.node_config = node_config
+        self.from_address = None
```
Comment (AgustinRamiroDiaz) [inline backend/node/genvm/base.py:20]: when is `None` a valid value for `from_address`?

**Review by AgustinRamiroDiaz — APPROVED** [review]
Changes look good and issue is addressed :+1:

File: backend/node/genvm/base.py (hunk 1 above)
Comment (cristiam86) [reply backend/node/genvm/base.py:18]: I don't think so :) The ContractRunner supports the execution of the IC code. The mode is just a property to understand how to execute some parts of the code and how to process the output of the execution.

File: backend/node/genvm/base.py [hunk 4]
```diff
@@ -13,6 +13,16 @@
...
+        self.gas_used = 0
+        self.eq_num = 0
+        self.eq_outputs = {"leader": {}}
```
[the remark this answers is not in the document]
Comment (cristiam86) [reply backend/node/genvm/base.py]: Yeah! let's talk about this later please it wasn't clear for me to add another package only for this and how to set the default value for the eq_outputs field

File: backend/node/genvm/base.py (hunk 2 above)
Comment (cristiam86) [reply backend/node/genvm/base.py:16]: Done!

File: backend/node/genvm/base.py (hunk 4 above)
[the remark this answers is not in the document]
Comment (AgustinRamiroDiaz) [reply backend/node/genvm/base.py]: No problem, just to be clear it's from the standard python's library

**Review by AgustinRamiroDiaz — APPROVED** [review]

File: backend/node/genvm/base.py (hunk 3 above)
Comment (cristiam86) [reply backend/node/genvm/base.py]: sorry, this was unanswered: when you need to get a contract schema or read from a contract state, you are not sending a transaction thus the from_address is None.
More info: https://stackoverflow.com/a/51850115/637623

**Pull request merged**